Add Kali-specific vars

jsf9k opened the pull request:
## 🗣 Description

In this pull request I use @mcdonnnj's Ansible `regex_replace()` foo and create a Kali-specific vars file.  We no longer let Kali "fall through" and use the Debian vars file.

## 💭 Motivation and Context

Kali recently dropped support for the python-pip package, which broke this Ansible role, so a change was necessary.

## 🧪 Testing

All pre-commit hooks and molecule tests pass.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

mcdonnnj committed:
Fixed broken hook id for pre-commit-terraform:
  terraform_validate_no_variables was changed to terraform_validate in the
  following commit:
  https://github.com/antonbabenko/pre-commit-terraform/commit/35e0356188b64a4c5af9a4e7200d936e514cba71.
Ran pre-commit autoupdate.

mcdonnnj committed:
Merge pull request #40 from cisagov/improvements/update_pre-commit

Update pre-commit Hooks and Fix Broken Hook ID

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

mcdonnnj committed:
Merge pull request #36 from cisagov/improvements/upstream_updates

Pull in Upstream Updates

mcdonnnj committed:
Ran `pre-commit autoupdate` to get the latest version of the
`pre-commit-terraform` hook since the PR I submitted,
https://github.com/antonbabenko/pre-commit-terraform/pull/100,
was approved. This will fix issues with `skeleton-tf-module` related to multiple
directories with Terraform code.

mcdonnnj committed:
Explicitly install pre-commit hooks as its own step so it's clearer when the
failure is with setup rather than hooks running.

mcdonnnj committed:
Rename action to accurately reflect that we are only linting, not building.
Adjust description for pre-commit hook setup to better convey what is being
done.

mcdonnnj committed:
Sort .gitignore entries.

mcdonnnj committed:
Merge pull request #41 from cisagov/bugfix/update_pre-commit_for_pre-commit-terraform_hook

Update pre-commit Hooks to Include Bugfix

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_upstream_updates

mcdonnnj committed:
Merge pull request #37 from cisagov/improvements/pull_upstream_updates

Pull in skeleton-generic Updates

jsf9k committed:
Add a Kali-specific vars file

Note that we also use @mcdonnnj's regex foo to convert the OS family
and distribution values for Kali into something that can actually be a
valid filename in Linux.

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into bugfix/add-kali-specific-vars

jsf9k commented:
Once this pull request is approved, I will create a similar pull requests for the following repos:
* ansible-role-amazon-efs-utils
* ansible-role-clamav
* ansible-role-cloudwatch-agent
* ansible-role-docker
* ansible-role-vnc-server
* ansible-role-openvpn
* ansible-role-python
* ansible-role-upgrade
* ansible-role-ufw

I think these repos (and only these repos) contain similar "fall-through to Debian" code.

dav3r commented on vars/Kali_GNU_Linux.yml:
```diff
@@ -0,0 +1,8 @@
+---
+# vars file for Kali
+
+# The pip package names.  These values are identical to those fro
```
fro —> from

jsf9k replied:
See commit 662febb.

felddy replied:
That what friends are fro!

dav3r reviewed (approved):
Looks good. 👍

mcdonnnj commented on vars/Kali_GNU_Linux.yml:
```diff
@@ -0,0 +1,8 @@
+
+# The pip package names.  These values are identical to those fro
+# Debian 11.
+package_names:
+  - python3-pip
+  - python3-dev
```
The `python-pip` package includes `python-pip-whl` for `pip` in a `venv` support ([per the Debian package description](https://packages.debian.org/unstable/python-pip-whl)). Since this was previously installed I feel like for consistency we should be installing it still.

mcdonnnj replied:
After the reply from @jsf9k I went to check out what the install would entail in a Docker instance of `kali-rolling`. When I installed the `python3-pip` package it also installed the `python-pip-whl` package. Which makes sense because as the [Debian package info page](https://packages.debian.org/unstable/python3-pip) shows, the `python-pip-whl` package is a dependency. Looks like I was a bit overzealous in my installation concerns.

jsf9k replied:
Although it spawned @mcdonnnj's insight, my comment was incorrect.  It appears that `python-pip-whl` is a python3 thang as well.

felddy replied:
***THANG!***

mcdonnnj commented on vars/Kali_GNU_Linux.yml:
```diff
@@ -0,0 +1,8 @@
+# vars file for Kali
+
+# The pip package names.  These values are identical to those fro
+# Debian 11.
```
Should this still say Debian 11?

jsf9k replied:
I think so.  Debian 11 is Bullseye.

mcdonnnj replied:
Ah, looking again I misread. Sorry for the false alarm.

mcdonnnj reviewed (approved):
Nice turnaround, please see my comments for consideration.

I am still unsure if the `python-pip` package was removed or if this is due to an error (there's a missing deb file compared to previous releases), but this will allow us to progress no matter what they're doing.

jsf9k commented:
I’m pretty sure python-pip-whl only makes sense for python2.

jsf9k committed:
Correct typo

Thanks to @dav3r for pointing this out.

felddy commented on tasks/main.yml:
```diff
@@ -6,14 +6,12 @@
+        # Add regex_replace() filters to remove spaces and
+        # slashes. This allows the OS family/distribution for Kali,
+        # "Kali GNU/Linux", to be easily mapped to a vars file.
+        - "{{ ansible_distribution | regex_replace('[ /]', '_') }}_{{ ansible_distribution_version }}.yml"
```
I like this sauce! 🐰

jsf9k replied:
It's @mcdonnnj's sauce, not mine.  Credit where credit is due.

jsf9k replied:
Also, that is a cute bunny.  Credit where credit is due.

felddy reviewed (approved):
Great job guys!  🐰 🥚 🌸 
![image](https://user-images.githubusercontent.com/3229435/79068201-28840980-7c93-11ea-9fa8-9d280371fe23.png)

jsf9k commented:
I think the upstream is still broken, since `python-pip` has been removed but python2 is still the default python.  We may still have to wait a bit for this to resolve.

felddy commented:
https://bugs.kali.org/view.php?id=6287

hillaryj committed:
Update CODEOWNERS

hillaryj committed:
Merge pull request #42 from cisagov/hillaryj-owner

Update CODEOWNERS

jsf9k committed:
Upgrade to molecule 3

To avoid a deprecation warning we rename playbook.yml file be renamed
to converge.yml.  The Dockerfile.j2 file is no longer used, so we
remove it.

jsf9k committed:
Add updated INSTALL.rst file

jsf9k committed:
Update molecule.yml for molecule 3

Molecule has greatly changed the linting configuration.  We also add
interpreter_python: auto, which makes Ansible go ahead and determine
the correct version of python to use using the heuristics that will
become the default in Ansible 2.12 anyway.  For more details, please
see this link:
https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upgrade-to-molecule-3

jsf9k committed:
Merge pull request #38 from cisagov/improvement/upgrade-to-molecule-3

Upgrade to molecule 3

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into bugfix/add-kali-specific-vars

jsf9k committed:
Add a custom Dockerfile for Debian 9 that forces python2

This is necessary to ensure the CyHy AMIs (which are the only ones
still built on Debian 9) continue to build as expected.  Once CyHy is
ported to python3 we can drop support for Debian 9.

jsf9k committed:
Merge pull request #39 from cisagov/bugfix/save_cyhy

Add a custom Dockerfile for Debian 9 that forces python2

jsf9k committed:
Modify vars files to only install pip2 for Debian 9

We treat Debian 9 as a special case, since the CyHy AMIs (and only the
CyHy AMIs) are built on it; therefore, Debian 9 requires python2
support.

jsf9k committed:
Update tests to agree with recent vars files changes

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into bugfix/add-kali-specific-vars

jsf9k committed:
Replace missing line

This must have been a typo on my part.

jsf9k committed:
Remove interpreter_python: auto

Molecule already enforces this option:
https://github.com/ansible-community/molecule/blob/master/molecule/provisioner/ansible.py#L392

jsf9k committed:
Merge pull request #40 from cisagov/bugfix/typo

Fix typo

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into bugfix/add-kali-specific-vars

dav3r reviewed (approved):
This all looks good- thanks for chasing everything down here! 🍔

jsf9k committed:
Add a symlink

Some Debian Stretch Docker containers are identified by Ansible as
version 9, others as version 9.12.

felddy reviewed (approved):
What a long strange trip it's been. 🌺 ☮️ 🌈 ✌️
